File: programmers/etc/level1/숫자_짝꿍.py
from collections import Counter

def solution(X, Y):
    answer = ""
    X, Y = Counter(X), Counter(Y)
    for i in range(9, -1, -1):
        cnt = min(X[str(i)], Y[str(i)])
        answer += str(i)*cnt
    if not answer:
        return "-1"
    elif answer[0] == "0":
        return "0"
    return answer


X, Y = "100", "2345"
print(solution(X, Y)) # "-1"

X, Y = "100", "203045"
print(solution(X, Y)) # "0"

X, Y = "100", "123450"
print(solution(X, Y)) # "10"

X, Y = "12321", "42531"
print(solution(X, Y)) # "321"

X, Y = "5525", "1255"
print(solution(X, Y)) # "552"

File: Java/programmers/etc/level1/숫자짝꿍.java
package CodeTest.Java.programmers.etc.level1;

public class 숫자짝꿍 {
    public String solution(String X, String Y) {
        int[] numX = new int[10], numY = new int[10];
        
        for (String x : X.split("")) {
            numX[Integer.parseInt(x)]++;
        }
        for (String y : Y.split("")) {
            numY[Integer.parseInt(y)]++;
        }
        
        StringBuilder tmp = new StringBuilder();

        for (int i=9; i>=0; i--) {
            if (numX[i] > 0 && numY[i] > 0) {
                tmp.append(String.valueOf(i).repeat(Math.min(numX[i], numY[i])));
            }
        }
        
        String answer = tmp.toString();
        
        return answer.startsWith("0") ? "0" : answer.equals("") ? "-1" : answer;
    }
}
